src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -479,10 +480,13 @@ static bool CallRegisterMethodByIndex (JniNativeMethodRegistrationArguments argu
 			}
 		}
 
-		public override void RegisterNativeMembers (JniType nativeClass, Type type, string? methods) =>
+		public override void RegisterNativeMembers (JniType nativeClass, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)] Type type, string? methods) =>
 			RegisterNativeMembers (nativeClass, type, methods.AsSpan ());
 
-		public void RegisterNativeMembers (JniType nativeClass, Type type, ReadOnlySpan<char> methods)
+		[UnconditionalSuppressMessage ("Trimming", "IL2057", Justification = "Type.GetType() can never statically know the string value parsed from parameter 'methods'.")]
+		[UnconditionalSuppressMessage ("Trimming", "IL2067", Justification = "Delegate.CreateDelegate() can never statically know the string value parsed from parameter 'methods'.")]
+		[UnconditionalSuppressMessage ("Trimming", "IL2072", Justification = "Delegate.CreateDelegate() can never statically know the string value parsed from parameter 'methods'.")]
+		public void RegisterNativeMembers (JniType nativeClass, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)] Type type, ReadOnlySpan<char> methods)
```

Comment:
Code-formatting wise, I think adding these parameter custom attributes makes the resulting code less readable, so I would prefer changing these into multi-line signatures:

```csharp
public void RegisterNativeMembers (
        JniType nativeClass,
        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)]
        Type type,
        ReadOnlySpan<char> methods)
{
    // …
}
```

Additionally, why `[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)]` and not `[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]`?

Reply:
Apparently the trimmer warning specifies which values to use.